lib/srv/alpnproxy/local_proxy.go

```diff
@@ -161,6 +161,14 @@ func NewLocalProxy(cfg LocalProxyConfig, opts ...LocalProxyConfigOpt) (*LocalPro
 
 // Start starts the LocalProxy.
 func (l *LocalProxy) Start(ctx context.Context) error {
+	if l.cfg.HTTPMiddleware != nil {
+		return trace.Wrap(l.StartHTTPAccessProxy(ctx))
+	}
+	return trace.Wrap(l.start(ctx))
+}
+
+// start starts the LocalProxy for raw TCP or raw TLS (non-HTTP) connections.
+func (l *LocalProxy) start(ctx context.Context) error {
 	if l.cfg.Middleware != nil {
 		err := l.cfg.Middleware.OnStart(ctx, l)
 		if err != nil {
```

Comment:
Something about this change isn't making sense. At the very least the `if l.cfg.Middleware != nil` clause in `start` should be removed because it will never be triggered.

Reply:
its `l.cfg.Middleware` here (not `l.cfg.HTTPMiddleware`)